### ISSUE TITLE
Only Left Eye Renders on VisionPro

### ISSUE THREAD

**Issue by mike-olson** [ninjamode/Unity-VR-Gaussian-Splatting#3]
I know VP was no listed on your device list - but it looks close! I'm using the URP integration and compiling to metal. Left eye seems to render ok, but the right eye has only the Unity Skybox - no splat. This would be super exciting if it ran on my VisionPro!

**Comment by ninjamode** [comment]
I believe the problem with the Vision Pro is that Unity required Instanced Rendering to be used with it, which is not supported as of now.

Specifically, looking at https://github.com/ninjamode/Unity-VR-Gaussian-Splatting/blob/928a76555ec2cc468cc51be9674aeb5c186b1eb9/package/Runtime/GaussianSplatRenderer.cs#L176 you’ll see a DrawProcedural call drawing all the Gaussians in the scene already as a instance of a triangle. To make this work with instanced Vr rendering, we would need to draw twice as many triangles, with the half of them adjusted for the right eye. That would need some shader adjustments.

Long story short: VR instanced rendering is not working as we already do some instancing. I do hope though that I can add support for it in the future!

**Comment by billzhao9** [comment]
Yes! Great work! hope this problem can be resolved.

**Comment by Aresnan** [comment]
> I know VP was no listed on your device list - but it looks close! I'm using the URP integration and compiling to metal. Left eye seems to render ok, but the right eye has only the Unity Skybox - no splat. This would be super exciting if it ran on my VisionPro!

Bro, are you success finally? If you do, please connect me.

**Comment by Aresnan** [comment]
> I know VP was no listed on your device list - but it looks close! I'm using the URP integration and compiling to metal. Left eye seems to render ok, but the right eye has only the Unity Skybox - no splat. This would be super exciting if it ran on my VisionPro!

Bro, did you finally succeed? If you did, please connect with me.

**Comment by Aresnan** [comment]
I solved this exellent problem, quite easy!!!

**Comment by yutaokuyama** [comment]
@Aresnan 
> I solved this exellent problem, quite easy!!!

I would greatly appreciate it if you could kindly share your solution. I am very interested in learning about its implementation.

**Comment by Aresnan** [comment]
> @Aresnan
> 
> > I solved this exellent problem, quite easy!!!
> 
> I would greatly appreciate it if you could kindly share your solution. I am very interested in learning about its implementation.

You mean, launch this project on vp properly?

**Comment by billzhao9** [comment]
> I solved this exellent problem, quite easy!!!

I also want to learn how you achieved it in Vision Pro. Would you like to share a repo? Appreciate it!

**Comment by Aresnan** [comment]
> [@Aresnan](https://github.com/Aresnan)
> 
> > I solved this exellent problem, quite easy!!!
> 
> I would greatly appreciate it if you could kindly share your solution. I am very interested in learning about its implementation.



> > I solved this exellent problem, quite easy!!!
> 
> I also want to learn how you achieved it in Vision Pro. Would you like to share a repo? Appreciate it!

call me

**Comment by yxz577** [comment]
> > [@Aresnan](https://github.com/Aresnan)
> > > I solved this exellent problem, quite easy!!!
> > 
> > 
> > I would greatly appreciate it if you could kindly share your solution. I am very interested in learning about its implementation.
> 
> > > I solved this exellent problem, quite easy!!!
> > 
> > 
> > I also want to learn how you achieved it in Vision Pro. Would you like to share a repo? Appreciate it!
> 
> call me

I want to learn how you achieved it in Apple Vision Pro, too! How can I call you? My homepage has my contact information, please contact me if it's convenient.Thank you！

**Comment by yxz577** [comment]
> > [@Aresnan](https://github.com/Aresnan)
> > > I solved this exellent problem, quite easy!!!
> > 
> > 
> > I would greatly appreciate it if you could kindly share your solution. I am very interested in learning about its implementation.
> 
> > > I solved this exellent problem, quite easy!!!
> > 
> > 
> > I also want to learn how you achieved it in Vision Pro. Would you like to share a repo? Appreciate it!
> 
> call me

@Aresnan I want to learn how you achieved it in Apple Vision Pro, too! How can I call you? My homepage has my contact information, please contact me if it's convenient.Thank you！

**Comment by ninjamode** [comment]
Development on this mostly happens in Aras' repository now, so ill close this.
